### PR TITLE
test report code

### PR DESCRIPTION
**Pull request by Yarden-Ingber**
add test results listener to post results to report

### CODE REVIEW

File: eyes.selenium.java/src/test/java/com/applitools/eyes/selenium/SendTestResultsListener.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        Map<String, ISuiteResult> suiteResultMap = iSuite.getResults();
+        Iterator<ISuiteResult> suiteResultIterator = suiteResultMap.values().iterator();
+        while (suiteResultIterator.hasNext()) {
+            ISuiteResult suiteResult = suiteResultIterator.next();
+            allTestResults.addAll(suiteResult.getTestContext().getPassedTests().getAllResults());
+            allTestResults.addAll(suiteResult.getTestContext().getFailedButWithinSuccessPercentageTests().getAllResults());
+            allTestResults.addAll(suiteResult.getTestContext().getFailedConfigurations().getAllResults());
+            allTestResults.addAll(suiteResult.getTestContext().getFailedTests().getAllResults());
+            allTestResults.addAll(suiteResult.getTestContext().getSkippedConfigurations().getAllResults());
+            allTestResults.addAll(suiteResult.getTestContext().getSkippedTests().getAllResults());
```
Comment (itaibh) [inline eyes.selenium.java/src/test/java/com/applitools/eyes/selenium/SendTestResultsListener.java]: I think `suitResult.getTestContext()` should be extracted to a local variable thus reducing the amount of code.

File: eyes.selenium.java/pom.xml [hunk 2]
```diff
@@ -42,6 +42,21 @@
...
+        <dependency>
+            <groupId>com.squareup.retrofit2</groupId>
+            <artifactId>converter-gson</artifactId>
+            <version>2.6.1</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.5</version>
+        </dependency>
```
Comment (itaibh) [inline eyes.selenium.java/pom.xml]: I'm not sure adding those dependencies is a good idea as they will be used by our customers...
@danielputerman what do you say?

File: eyes.selenium.java/src/test/java/com/applitools/eyes/selenium/SendTestResultsListener.java (hunk 1 above)
Comment (Yarden-Ingber) [reply eyes.selenium.java/src/test/java/com/applitools/eyes/selenium/SendTestResultsListener.java]: changed. thanks!

File: eyes.selenium.java/pom.xml (hunk 2 above)
Comment (myosher1) [reply eyes.selenium.java/pom.xml]: @Yarden-Ingber - Itai is right we are using Jackson for parsing JSONs...

Comment (Yarden-Ingber) [reply eyes.selenium.java/pom.xml]: @itaibh @myosher1 - I can change to jackson, but should I also not use retrofit?

**Pull request closed without merging**